### PR TITLE
fix some more risky tests

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

Continuation of #22066 fixing the tests for the following components:

* Config
* Form
* HttpFoundation
* Security

### CODE REVIEW

File: src/Symfony/Component/Form/Tests/SimpleFormTest.php [hunk 1]
```diff
@@ -1036,15 +1037,17 @@ public function testPostSubmitDataIsNullIfInheritData()
...
-                $test->fail('The SUBMIT event should not be fired');
+            ->addEventListener(FormEvents::SUBMIT, function (FormEvent $event) use (&$called) {
+                ++$called;
             })
             ->setInheritData(true)
             ->getForm();
 
         $form->submit('foo');
+
+        $this->assertSame(0, $called);
```
Comment (stof) [inline src/Symfony/Component/Form/Tests/SimpleFormTest.php]: I would add a message here (``'The SUBMIT event should not be fired.'``) to make the failure easier to understand

File: src/Symfony/Component/Form/Tests/SimpleFormTest.php [hunk 2]
```diff
@@ -933,6 +932,8 @@ public function testSubmittingWrongDataIsIgnored()
             ->getForm();
 
         $parent->submit('not-an-array');
+
+        $this->assertSame(0, $called);
```
Comment (stof) [inline src/Symfony/Component/Form/Tests/SimpleFormTest.php]: a message should be added

File: src/Symfony/Component/Security/Acl/Tests/Dbal/MutableAclProviderTest.php [hunk 3]
```diff
@@ -401,17 +402,25 @@ public function testUpdateAclDeletingObjectFieldAcesThrowsDBConstraintViolations
...
         $acl->insertObjectFieldAce($fieldName, $sid2, 4);
         $provider->updateAcl($acl);
 
+        $this->assertCount(2, $provider->findAcl($oid)->getObjectFieldAces($fieldName));
+
         $index = 0;
         $acl->deleteObjectFieldAce($index, $fieldName);
         $provider->updateAcl($acl);
 
+        $this->assertCount(1, $provider->findAcl($oid)->getObjectFieldAces($fieldName));
```
Comment (stof) [inline src/Symfony/Component/Security/Acl/Tests/Dbal/MutableAclProviderTest.php]: shouldn't this be split into multiple tests instead ? Having assertions at 4 different points of the test looks weird.

File: src/Symfony/Component/Security/Core/Tests/Encoder/BCryptPasswordEncoderTest.php [hunk 4]
```diff
@@ -42,7 +42,7 @@ public function testCostAboveRange()
     public function testCostInRange()
     {
         for ($cost = 4; $cost <= 31; ++$cost) {
```
Comment (stof) [inline src/Symfony/Component/Security/Core/Tests/Encoder/BCryptPasswordEncoderTest.php]: this should use a data provider, to ensure that 31 is tested even if we have a failure for 4

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Form/Tests/FormConfigTest.php [hunk 5]
```diff
@@ -24,72 +22,65 @@ class FormConfigTest extends TestCase
...
-            array("a\t", false),
-            array("a\n", false),
-            array('a-', true),
-            array('a_', true),
-            array('a:', true),
+            array('z0'),
+            array('A0'),
+            array('A9'),
+            array('Z0'),
+            array('#', '\Symfony\Component\Form\Exception\InvalidArgumentException'),
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Tests/FormConfigTest.php]: do we really need the leading backslash?

**Review by nicolas-grekas — APPROVED** [review]
:+1:

**Review by stof — APPROVED** [review]
👍

**Comment by fabpot** [comment]
Wow, you should have spend a lot of time on this one. Thanks @xabbuh.

**Pull request merged**